optuna/pruners/percentile.py

```diff
@@ -93,10 +116,13 @@ def prune(self, study, trial):
         if step is None:
             return False
 
-        if step <= self.n_warmup_steps:
+        n_warmup_steps = self.n_warmup_steps
+        if step <= n_warmup_steps:
             return False
 
-        if len(trial.intermediate_values) == 0:
+        if not _is_first_step_after_nearest_lower_pruning_step(
```

Comment:
Please add a comment to clarify this check is for `interval_steps`.
The method name does not contain `interval_steps`, and the future developers may not understand the relationship between this method and the `interval_steps` at first glance.

Reply:
That's a good point. I changed the name in https://github.com/pfnet/optuna/pull/660/commits/dea7a26a5b166edcd319ac556d9858835dd9ae2c.